pkg/cmd/kafka/topic/consume/consume.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// NewComsumeTopicCommand creates a new command for producing to a kafka topic.
+func NewConsumeTopicCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "consume",
+		Short:   f.Localizer.MustLocalize("kafka.topic.consume.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("kafka.topic.consume.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("kafka.topic.consume.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if opts.kafkaID == "" {
+
+				kafkaInstance, err := contextutil.GetCurrentKafkaInstance(f)
+				if err != nil {
+					return err
+				}
+
+				opts.kafkaID = kafkaInstance.GetId()
+			}
+
+			return runCmd(opts)
+		},
+	}
+
+	flags := kafkaflagutil.NewFlagSet(cmd, f.Localizer)
+
+	flags.StringVar(&opts.topicName, "name", "", f.Localizer.MustLocalize("kafka.topic.common.flag.name.description"))
+	flags.Int32Var(&opts.partition, "partition", 0, f.Localizer.MustLocalize("kafka.topic.consume.flag.partition.description"))
+	flags.StringVar(&opts.from, "from", DefaultTimestamp, f.Localizer.MustLocalize("kafka.topic.consume.flag.from.description"))
+	flags.BoolVar(&opts.wait, "wait", false, f.Localizer.MustLocalize("kafka.topic.consume.flag.wait.description"))
+	flags.Int64Var(&opts.offset, "offset", DefaultOffset, f.Localizer.MustLocalize("kafka.topic.consume.flag.offset.description"))
+	flags.Int32Var(&opts.limit, "limit", DefaultLimit, f.Localizer.MustLocalize("kafka.topic.consume.flag.limit.description"))
+	flags.StringVar(&opts.outputFormat, "format", "json", f.Localizer.MustLocalize("kafka.topic.consume.flag.format.description"))
+
+	_ = cmd.MarkFlagRequired("name")
+
+	_ = cmd.RegisterFlagCompletionFunc("name", func(cmd *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return kafkacmdutil.FilterValidTopicNameArgs(f, toComplete)
+	})
+
+	flags.AddInstanceID(&opts.kafkaID)
+	flagutil.EnableStaticFlagCompletion(cmd, "format", outputFormatTypes)
+
+	return cmd
+}
+
+func runCmd(opts *options) error {
+
+	conn, err := opts.f.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, _, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	if opts.wait {
+
+		if opts.limit != DefaultLimit {
+			opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.topic.consume.log.info.limitIgnored", localize.NewEntry("Limit", DefaultLimit)))
+			opts.limit = DefaultLimit
+		}
+
+		if opts.offset != DefaultOffset {
+			opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.topic.consume.log.info.offsetIgnored", localize.NewEntry("Offset", DefaultOffset)))
+			opts.offset = DefaultOffset
+		}
+
+		if opts.from == DefaultTimestamp {
+			// get current time in ISO 8601
+			opts.from = time.Now().Format(time.RFC3339)
+		}
+
+		max_offset := opts.offset
+		for true {
```

Comment:
I think we look for equivalent of thread in java or set timeout.
Check go routines. Endless loop does work but we need logic to wait 
to not get rate limited

Reply:
I checked go routines and they seems to not be worth it, it always ended up being that we just wait on the separate thread to end. Adding some kind of sleep that is less than the rate limit is what I'll do.

Reply:
Added sleep in for loop to not hit rate limit